streampipes-extensions/streampipes-connect-adapters-iiot/src/main/java/org/apache/streampipes/connect/iiot/adapters/iolink/IfmAlMqttAdapter.java

```diff
@@ -110,18 +110,26 @@ public void onAdapterStarted(IAdapterParameterExtractor extractor,
 
               for (int i = 0; i < ports.size(); i++) {
 
-                var portResult = getMap(payload,
-                    "/iolinkmaster/port[%s]/iolinkdevice/pdin".formatted(ports.get(i)));
-                var eventData = (String) portResult.get("data");
-
-                var parsedEvent = sensor.parseEvent(eventData);
-                parsedEvent.put("timestamp", System.currentTimeMillis() + i);
-                parsedEvent.put("port", "port" + ports.get(i));
-                parsedEvent.put(SensorVVB001.IO_LINK_MASTER_SN, serialnumber);
-
-                collector.collect(parsedEvent);
+                try {
+                  var portResult = getMap(payload,
+                          "/iolinkmaster/port[%s]/iolinkdevice/pdin".formatted(ports.get(i)));
```

Comment:
Can we check if the port exists here before fetching it?

Reply:
Done